CarDealer/CarBuilderDirector.cs

```diff
@@ -0,0 +1,30 @@
+﻿using CarDealer.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarDealer
+{
+    class CarBuilderDirector
+    {
+        public ICarBuilder builder { get; set; }
+        public void Construct(Model model, Color color, Engine engine)
+        {
+            builder.SetModel(model);
+            builder.SetColor(color);
+            builder.SetEngine(engine);
+        }
+        public CarBuilderDirector()
```

Comment:
Aici pe constructor primesti un ICarBuilder, uita-te pe diagrama ...
ai ctor(ICarBuilder builder)
{
      this.builder = builder;
}